app/models/concerns/slack_message_creation.rb

```diff
@@ -0,0 +1,69 @@
+module SlackMessageCreation
+  extend ActiveSupport::Concern
+
+  private
+
+  def create_message
+    return unless conversation
+
+    build_message
+    @message.save!
+    { status: 'success' }
+  rescue Slack::Web::Api::Errors::MissingScope => e
+    handle_missing_scope_error(e)
+  end
+
+  def handle_missing_scope_error(_error)
+    Rails.logger.error e
+    integration_hook.prompt_reauthorization!
+    integration_hook.disable
+  end
```

Comment:
This shouldn't be handled like this, instead. Change this function to handle only disabling the hook

```suggestion
  def disable_and_reauthorize
    integration_hook.prompt_reauthorization!
    integration_hook.disable
  end
```
and

```
  def create_message
    return unless conversation

    build_message
    @message.save!
    { status: 'success' }
  rescue Slack::Web::Api::Errors::MissingScope => e
    Rails.logger.error e
    disable_and_reauthorize
  end
```

Reply:
Also, we can use `ChatwootExceptionTracker`

Reply:
@scmmishra Done